src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -597,7 +656,7 @@ private BoundLambda ReallyBind(NamedTypeSymbol delegateType)
             }
             else
             {
-                lambdaSymbol = CreateLambdaSymbol(Binder.ContainingMemberOrLambda, returnType, cacheKey.ParameterTypes, cacheKey.ParameterRefKinds, refKind);
+                lambdaSymbol = CreateLambdaSymbol(Binder.ContainingMemberOrLambda!, returnType, cacheKey.ParameterTypes, cacheKey.ParameterRefKinds, refKind);
```

Comment:
>Binder.ContainingMemberOrLambda! [](start = 50, length = 32)

nit: should be an assertion

Reply:
Added an assertion to `UnboundLambda..ctor()`.